models/QtiRunnerInitDataBuilder.php

```diff
@@ -268,4 +263,29 @@ private function getServiceContext($deliveryExecutionId): QtiRunnerServiceContex
 
         return $this->qtiRunnerService->getServiceContext($testId, $compilation, $deliveryExecutionId);
     }
+
+    /**
+     * @param array $state
+     * @return int
+     */
+    private function getResponseCountFromState(array $state): int
+    {
+        $answers = 0;
+        if (count($state)) {
+            foreach ($state as $response) {
+                $responseCount = 0;
+
+                if (!empty($response['response']['base'])) {
+                    $responseCount = 1;
+                }
+                elseif (!empty($response['response']['list'])) {
+                    $responseCount = count(array_merge([], ...array_values($response['response']['list'])));
```

Comment:
why do we need `[]`?

Reply:
btw if you know that nesting level = 2 you can do the trick:
```
$responseCount = count($response['response']['list'], true) - count($response['response']['list'])
```

Reply:
We could have deeper nested level. Example
```
$itemState = [
   "RESPONSE" => [
         "response" => [
            "list" => [
               "identifier" => [] 
            ] 
         ] 
      ], 
   "RESPONSE_1" => [
                     "response" => [
                        "base" => null 
                     ] 
                  ] 
]; 
```
or 
```
<?php 

 $itemState = [
   "response" => [
         "list" => [
            "directedPair" => [
               [
                  "choice_1", 
                  "choice_4" 
               ], 
               [
                     "choice_2", 
                     "choice_4" 
                  ] 
            ] 
         ] 
      ] 
]; 
```